chia/_tests/wallet/sync/test_wallet_sync.py

```diff
@@ -1335,12 +1337,15 @@ async def new_func(puzzle_hash: bytes32) -> Optional[WalletIdentifier]:
 
         return new_func
 
-    with monkeypatch.context() as m:
-        m.setattr(
-            full_node_api,
-            "request_puzzle_solution",
-            flaky_request_puzzle_solution(full_node_api.request_puzzle_solution),
+    with contextlib.ExitStack() as exit_stack:
+        exit_stack.enter_context(
+            patch_request_handler(
+                api=FullNodeAPI,
+                handler=flaky_request_puzzle_solution(FullNodeAPI.request_puzzle_solution),
```

Comment:
Is this intentionally not `full_node_api.request_puzzle_solution` ?

Reply:
I started by just having this patch the class.  This can be problematic in tests where we have multiple of the same node type and only want one of them patched.  So I adjusted to support patching instances as well.  Thanks for catching these missed leftovers from that intermediate state.

One of the changes in this refactor is that the api is no longer handled with bound methods on the instance, but instead by the unbound methods on the class.  When they are called through the api stuff the `self` is passed explicitly now since it isn't bound anymore.

Now that I've written what I'm confident about, I'm not entirely sure what is better here.  As the framework is coded now, when patching you should be thinking in terms of functions that accept `self`.  But, that's not strictly required here since there's the additional wrapping layer provided by `flaky_request_puzzle_solution()` which can choose to call whatever it wants (in this case what it is passed).  I feel there is some value in the consistency here of taking the unbound method as coded.

I guess I'll go with this reasoning and preference.  Not intentional, a leftover of the evolution of the PR.  But, I think it might still be the better option.